CMakeLists.txt

```diff
@@ -58,6 +58,8 @@ option(UNSAFE_BUILD "Allow unsafe builds. These builds ARE NOT SECURE." OFF)
 option(MBEDTLS_FATAL_WARNINGS "Compiler warnings treated as errors" ON)
 if(CMAKE_HOST_WIN32)
     option(GEN_FILES "Generate the auto-generated files as needed" OFF)
+elseif(CMAKE_BUILD_TYPE STREQUAL "Release")
```

Comment:
As discussed: no, this is not what the task is about. The `Release` build type is about doing a release build vs a debug build (with optimization vs with debug symbols). The task is about a release of the project, as in, grabbing a tagged `mbedtls-3.y.z` or a zip of that. When taking the source code from a release, `GEN_FILES` should default off. When taking the source code from the development branch, `GEN_FILES` should default on.